semeio/workflows/localisation/local_script_lib.py

```diff
@@ -1,42 +1,98 @@
 # pylint: disable=W0201
-import yaml
+import itertools
 import math
+from collections import defaultdict
+from typing import List
+
+import yaml
 import cwrap
 import numpy as np
 
 from ecl.util.geometry import Surface
 from res.enkf.enums.ert_impl_type_enum import ErtImplType
 from res.enkf.enums.enkf_var_type_enum import EnkfVarType
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 
 from semeio.workflows.localisation.localisation_debug_settings import (
     LogLevel,
     debug_print,
 )
 import semeio.workflows.localisation.localisation_debug_settings as log_level_setting
 
-# The global variables are only used for controlling output of log info to screen
-# and for writing scaling factor to file (for QC and test purpose)
+# The global variable is only used for controlling output of log info to screen
 debug_level = log_level_setting.debug_level
-scaling_parameter_number = log_level_setting.scaling_parameter_number
 
 
-def get_observations_from_ert(ert):
-    # Get list of observation nodes from ERT config to be used in consistency check
-    ert_obs = ert.getObservations()
-    obs_data_from_ert_config = ert_obs.getAllActiveLocalObsdata()
-    ert_obs_node_list = []
-    for obs_node in obs_data_from_ert_config:
-        node_name = obs_node.key()
-        ert_obs_node_list.append(node_name)
-    return ert_obs_node_list
+@dataclass
+class Parameter:
+    name: str
+    type: str
+    parameters: List = field(default_factory=list)
+
+    def to_list(self):
+        if self.parameters:
+            return [f"{self.name}:{parameter}" for parameter in self.parameters]
+        else:
+            return [f"{self.name}"]
+
+    def to_dict(self):
+        return {self.name: self.parameters}
+
+
+@dataclass
+class Parameters:
```

Comment:
I wanted to change this into a self contained class which was able to convert between the list and dict formats.

Reply:
ok